src/nvim/mark.c

```diff
@@ -378,8 +378,8 @@ pos_T *getmark_buf_fnum(buf_T *buf, int c, int changefile, int *fnum)
     }
   } else if (ASCII_ISLOWER(c)) {      /* normal named mark */
     posp = &(buf->b_namedm[c - 'a']);
-  } else if (ASCII_ISUPPER(c) || VIM_ISDIGIT(c)) {    /* named file mark */
-    if (VIM_ISDIGIT(c))
+  } else if (ASCII_ISUPPER(c) || isdigit(c)) {    /* named file mark */
```

Comment:
The conditionals here looks kind of strange, this is out of the scope of this PR, but maybe note this for later:

isdigit(c) is checked twice if "c" is a digit, also a manual subtraction is used instead of something like "tolower()". (did not evaluate this in detail, but is just jumped into my face when looking over this)